src/librustc/ty/layout.rs

```diff
@@ -541,6 +545,9 @@ pub struct Struct {
     pub memory_index: Vec<u32>,
 
     pub min_size: Size,
+
+    /// Size of padding needed by the struct due to requested repr alignment
+    pub padding: Vec<Size>,
```

Comment:
You comment on this later so I guess you are OK with it?

It could be done on demand, but I thought it would be more overhead as it's needed in a couple of places (https://github.com/bitshifter/rust/blob/6b9c5d1ff6df1e0d6d181fb5049c3108359b47ee/src/librustc_trans/adt.rs#L265-L266 and https://github.com/bitshifter/rust/blob/6b9c5d1ff6df1e0d6d181fb5049c3108359b47ee/src/librustc/ty/layout.rs#L775-L776).

If we keep the vector I was thinking of putting it in an Option, since I think most people won't need `repr(align)` so it would generally be empty.